src/api/delivery-needed/index.js

```diff
@@ -109,3 +122,95 @@ exports.deliveryNeededRequestHandler = async (req, res) => {
     },
   });
 };
+
+/*
+ * This handler takes a delivery request code and the phone number of a delivery volunteer
+ * and assigns the volunteer to the delivery request. It also pings our Twilio service which then
+ * texts the delivery and intake volunteers with relevant information to initiate a
+ * delivery.
+ */
+exports.assignDeliveryHandler = async (req, res) => {
+  const { requestCode, phoneNumber } = req.body;
+
+  if (!requestCode || !phoneNumber)
+    throw new Error("Expected `requestCode` and `phoneNumber` in payload.");
+
+  let request;
+  let volunteer;
+  let error;
+
+  // validate request
+  try {
+    [request, error] = await findRequestByCode(requestCode);
+    if (error)
+      return res.status(404).send({
+        message: `${requestCode} not found.`,
+      });
+
+    if (request.fields.Status !== status_options.deliveryNeeded)
+      throw new Error(
+        `Cannot claim delivery ${requestCode} with status ${request.fields.Status}.`
+      );
+  } catch (e) {
+    return res.status(400).send({ message: e });
+  }
+
+  // find volunteer record in volunteer table
+  try {
+    [volunteer, error] = await findVolunteerByPhone(phoneNumber);
+
+    // TODO: on not found error, send volunteer form
+    if (!volunteer || (error && error.includes("404")))
+      return res.status(404).send({
+        message: error,
+        notFound: "volunteer",
+      });
+    if (error) throw new Error(error);
+  } catch (e) {
+    return res.status(400).send({
+      message: `Something went wrong while looking for volunteer record: ${e}`,
+    });
+  }
+
+  // assign delivery volunteer to request
+  try {
+    // use email username in place of an empty volunteer name field
+    // return empty string if neither exist
+    const {
+      [volunteersFields.name]: volunteerName,
+      [volunteersFields.email]: volunteerEmail,
+    } = volunteer.fields;
+    const name =
+      volunteerName || (volunteerEmail && volunteerEmail.split("@")[0]) || "";
+
+    // assign delivery to volunteer in airtable
+    [request, error] = await updateRequestByCode(requestCode, {
```

Comment:
the twilio webhook is also [doing the assigning](https://github.com/mab253/mutual-aid-app/blob/mab-sms-delivery/src/twilio/functions/sms/test-delivery.js#L37), (i posted about it belatedly on the [issue](https://github.com/crownheightsaid/mutual-aid-app/issues/123#issuecomment-676558469)). it makes sense for assigning to happen here in the API since we already have this `updateRequestByCode` helper, so we should remove the twilio assigning part to avoid accidental race conditions.

Reply:
sounds good to me! i can take this out of the Twilio function, no prob